aptos-indexer-processors-sdk/testing-framework/src/new_test_context.rs

```diff
@@ -64,18 +64,15 @@ impl SdkTestContext {
     pub async fn run<F>(
         &mut self,
         processor: &impl ProcessorTrait,
-        db_url: &str,
         txn_version: u64,
         generate_files: bool, // flag to control file generation
         output_path: String,  // output path
-        verification_f: F,    // Modified to return a HashMap for multi-table data
+        is_scenario_testing: bool,
```

Comment:
What's the context around this bool?

Reply:
for scenario testing which requires more customized validation, I didn't think we would have to generate expected db output files rather they can do manual checks on fields they are interested in. 

so this flag will be used to differentiate the behavior.

Reply:
How come you can't generate db outputs with multiple txn's? For multiple rows in a table, the expected db output could be an array of rows. I think we should allow generating db outputs and also let them write more custom validation if they want.

Reply:
good point, i was thinking, with multiple txns, only thing you would check is a few fields. 
then for the the return type, we could change it to Vec<Value>. this hsould support single txn handling then. 
` anyhow::Result<HashMap<String, Vec<serde_json::Value>>>`

Reply:
Yeah we probably need to make sure there's no collisions in the generate db output filename if you use the same transaction for multiple tests.

Could you also add a comment on the types that verification_f expects?

Reply:
yeah maybe I can add a optional param for file_name?

Reply:
That works! Also I just thought we don't need the flag, bc they can also choose to return empty HashMap from verification_f

Reply:
yeah if we are going to allow file generations for multiple txns, we don't need a flag here.

Reply:
thinking about it more, I think we can just keep `anyhow::Result<HashMap<String, serde_json::Value>>` as we can  fit vec of rows into one json value